base_delivery_carrier_label/models/stock_picking.py

```diff
@@ -85,23 +74,40 @@ def get_shipping_label_values(self, label):
             'file_type': label['file_type'],
         }
 
-    @api.multi
-    def generate_labels(self, package_ids=None):
-        """ Generate the labels.
+    def generate_labels(self):
+        """ Legacy method. Remove me after 12.0
+        """
+        _logger.warning(
+            "Your delivery module depending on your carrier must call "
+            "action_generate_carrier_label() method "
+            "instead of generate_labels()")
+        return self.action_generate_carrier_label()
+
+    def _set_a_default_package(self):
+        """ Pickings using this module must have a package
+            If not this method put it one silently
+        """
+        for picking in self:
+            move_lines = picking.move_line_ids.filtered(
+                lambda s: s.package_id and not s.result_package_id)
+            for move_line in move_lines:
+                move_line.write({'result_package_id': move_line.package_id.id})
```

Comment:
I don't understand this part...
Why write result_package_id if a move line with package id is found?
Seems very strange to me to have a package_id and a result_package_id with the same package on the same move line...
And I don't get why you have to do this and no simply something like this : 
`move_lines = picking.move_line_ids.filtered( lambda s: not (s.package_id or s.result_package_id))
if move_lines:
    package = self.env['stock.quant.package'].create({})
    move_lines.write({'result_package_id': package.id})
`